crates/ruff_server/src/format.rs

```diff
@@ -24,9 +32,14 @@ pub(crate) fn format_range(
 ) -> crate::Result<PrintedRange> {
     let format_options = formatter_settings.to_format_options(source_type, document.contents());
 
-    Ok(ruff_python_formatter::format_range(
-        document.contents(),
-        range,
-        format_options,
-    )?)
+    match ruff_python_formatter::format_range(document.contents(), range, format_options) {
+        Ok(formatted) => Ok(formatted),
+        // Special case - syntax/parse errors should be handled here instead of
+        // being propagated to become visible server errors.
+        Err(FormatModuleError::ParseError(error)) => {
+            tracing::warn!("Unable to format document range: {error}");
+            Ok(PrintedRange::new(document.contents().to_string(), range))
```

Comment:
Should we just return `Ok(None)` here, so that we avoid cloning the source text and applying a full no-op edit?

Reply:
Sure, I can re-work the return type to be optional 👍